include/cute/atom/copy_traits_xe.hpp

```diff
@@ -38,58 +38,147 @@
+                                                            get<1>(typename T::ValueShape{})
+                                                                / Int<detail::subgroup_size>{})));
+};
+
+
+// ==========  is_transpose_load  ==========
+template <class, class = void>
+static constexpr bool has_is_transpose = false;
+
+template <class T>
+static constexpr bool has_is_transpose<T, cute::void_t<decltype(T::is_transpose)>> = true;
 
-  template<>
-  struct is_transpose<XE_2D_U16x16x8_LD_T> : bool_constant<true>{};
+template <class, class = void>
+static constexpr bool is_transpose_load = false;
 
-  template<>
-  struct is_transpose<XE_2D_U16x16x16_LD_T> : bool_constant<true>{};
+template <class T>
+static constexpr bool is_transpose_load<T, std::enable_if_t<has_is_transpose<T>>> = T::is_transpose;
 
-  template<>
-  struct is_transpose<XE_2D_U32x16x2_LD_T> : bool_constant<true>{};
 
-  template<>
-  struct is_transpose<XE_2D_U32x16x4_LD_T> : bool_constant<true>{};
+// ==========  is_stride_leftmost  ==========
+template <class T, class = void>
+static constexpr bool is_stride_leftmost = std::is_same_v<_1, decltype(get<0>(T{}))>;
 
-  template<>
-  struct is_transpose<XE_2D_U32x16x8_LD_T> : bool_constant<true>{};
+template <class T>
+static constexpr bool is_stride_leftmost<T, cute::void_t<decltype(T{}.stride())>> = std::is_same_v<_1, decltype(get<0>(T{}.stride()))>;
 
-  template<>
-  struct is_transpose<XE_2D_U64x8x1_LD_T> : bool_constant<true>{};
 
-  template<>
-  struct is_transpose<XE_2D_U64x8x2_LD_T> : bool_constant<true>{};
+} // end namespace detail
+
+
+template <uint32_t dim, class Tensor_t>
+static constexpr auto append_pvc_tensor(Tensor_t const &t0, uint32_t shape, uint32_t stride) {
+  return make_tensor(make_inttuple_iter(t0.data()), append(t0.layout(), make_layout(shape, E<dim>{} * stride)));
+}
+
+template <class CopyOp, class StrideIndicator = cute::Stride<int64_t, cute::Int<1>, int64_t>>
+struct XE_2D_LD_Unpack {
 
-  template<>
-  struct is_transpose<XE_2D_U64x8x4_LD_T> : bool_constant<true>{};
+  using BlockShape = CopyOp::BlockShape;
+  using Value_Layout = typename detail::value_layout_t<CopyOp>::type;
+  using Traits_LD_t = Copy_Traits<CopyOp, StrideIndicator>;
 
-  template <class, class Enable = void> constexpr bool has_inst_dtype = false;
+  static constexpr auto stride_rank = rank(StrideIndicator{});
+  static_assert(stride_rank == 2 || stride_rank == 3);
 
-  template <class T>
-  constexpr bool has_inst_dtype<T, cute::void_t<typename T::inst_dtype>> = true;
-} // namespace detail end
+  // Assume LD_T/LD_N will indicate ColumnMajor and RowMajor
+  static constexpr bool is_column_major = detail::is_transpose_load<CopyOp>;
 
-template <class CopyOp, class... ArgTs> struct XE_2D_LD_Unpack {
+  // We need reverse some parameters becasue intel xe 2d copy intrinsic always assume the matrix is (M, N):(N, 1) convention
+  static constexpr bool is_need_reversed = detail::is_stride_leftmost<StrideIndicator>;
```

Comment:
This assumes that if it isn't LeftLayout it is RightLayout. Any such assumptions we should check with static_assert. E.g. by first checking that at least one of of the m,n strides are 1.